### PR TITLE
Init Poser v2.0

### LINKED ISSUE #31: New major version?
I guess it's time to raise requirements, allow some new features (e.g. Symfony 5) and drop unsupported deps

### PR DESCRIPTION
**Pull request by JellyBellyDev**
This is a feature branch for Poser v2.0: 

- [x] create a feature branch `release-v2`
- [x] upgrade docker-compose for dev environment with php7.4
- [x] upgrade php to 7.4 drop unsupported deps
- [x] add CircleCI for build and tests
- [x] upgrade readme
- [x] refactoring library to full typed
- [x] configure and run php-cs-fixer
- [x] rename SvgRender class to SvgPlasticRender to match the same naming pattern with other render classes
- [x] phpstan?
- [ ] merge and tag v2.0.0

close #31

### CODE REVIEW

File: spec/PUGX/Poser/PoserSpec.php [hunk 1]
```diff
@@ -37,15 +37,15 @@ function it_should_be_able_to_generate_an_svg_image_from_URI()
     }
 
 
-    public function getMatchers()
+    public function getMatchers() :array
```
Comment (AlessandroMinoccheri) [inline spec/PUGX/Poser/PoserSpec.php:40]: This
```
) :array
```

should be 

```
): array
```

File: src/UI/Command.php [hunk 2]
```diff
@@ -101,6 +101,8 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $this->printHeaderOnce($output);
             throw $e;
         }
+
+        return 0;
```
Comment (AlessandroMinoccheri) [inline src/UI/Command.php:105]: Is it necessary?

**Comment by AlessandroMinoccheri** [comment]
What do you think to try psalm instead of phpstan?

File: src/UI/Command.php (hunk 2 above)
Comment (JellyBellyDev) [reply src/UI/Command.php:105]: Yep! The new version of symfony command require an exit code!

**Comment by JellyBellyDev** [comment]
> What do you think to try psalm instead of phpstan?

I've never tried it! It could be the right occasion! ;)

File: src/UI/Command.php (hunk 2 above)
Comment (AlessandroMinoccheri) [reply src/UI/Command.php:105]: Well, we should add return type hint in this case or is not necessary?

Comment (JellyBellyDev) [reply src/UI/Command.php:105]: It’s a bullet of the roadmap! ;)

**Comment by zerai** [comment]
It's a minor relevance question, 
can we move the color validation before the assignment and escape operations?
Badge class
```php
    public function __construct($subject, $status, $color, $format = self::DEFAULT_FORMAT)
    {
        $this->subject = $this->escapeValue($subject);
        $this->status  = $this->escapeValue($status);
        $this->format  = $this->escapeValue($format);
        $this->color   = $this->getColorHex($color);

        if (!$this->isValidColorHex($this->color)) {
            throw new \InvalidArgumentException(sprintf('Color not valid %s', $this->color));
        }
    }
```

**Comment by garak** [comment]
> It's a minor relevance question,
> can we move the color validation before the assignment and escape operations?
> Badge class
> 
> ```
>     public function __construct($subject, $status, $color, $format = self::DEFAULT_FORMAT)
>     {
>         $this->subject = $this->escapeValue($subject);
>         $this->status  = $this->escapeValue($status);
>         $this->format  = $this->escapeValue($format);
>         $this->color   = $this->getColorHex($color);
> 
>         if (!$this->isValidColorHex($this->color)) {
>             throw new \InvalidArgumentException(sprintf('Color not valid %s', $this->color));
>         }
>     }
> ```

I would move the check in `getColorHex` method instead

**Comment by JellyBellyDev** [comment]
> > It's a minor relevance question,
> > can we move the color validation before the assignment and escape operations?
> > Badge class
> 
> I would move the check in `getColorHex` method instead

I approve to move the validation into `getColorHex`! ;)

**Comment by zerai** [comment]
If you move the validation, basically, you accept every string (x@^?) as a color.
Is it a valid/usable Badge? or I'm missing your pov

**Comment by garak** [comment]
> If you move the validation, basically, you accept every string (x@^?) as a color.

No. Validation is just moved, but still applies.

**Comment by antonkomarev** [comment]
How about rename `SvgRender` class to `SvgPlasticRender` to match the same naming pattern with other render classes?

**Comment by JellyBellyDev** [comment]
> How about rename `SvgRender` class to `SvgPlasticRender` to match the same naming pattern with other render classes?
> It was the only one not obvious thing when I started to use Poser.

Good idea @antonkomarev! Can you make a PR?

> P.S.: Thanks a lot for this package! It was really helpful when I decided to add some customization features to [GitHub profile views counter](https://github.com/antonkomarev/github-profile-views-counter).

Your welcome!!! 🚀

**Comment by JellyBellyDev** [comment]
@garak @AlessandroMinoccheri what do you think about if we give up phpstan or psalm and publish this v2.0? :)

**Comment by antonkomarev** [comment]
@JellyBellyDev I'd prefer to refactor `style` & `format` things before v2 release, otherwise we will be able to do it only in v3.

**Comment by garak** [comment]
I agree with Anton. Let's make a little further step

**Comment by JellyBellyDev** [comment]
> @JellyBellyDev I'd prefer to refactor `style` & `format` things before v2 release, otherwise we will be able to do it only in v3.
> 
> I've described motivation here: [#41 (comment)](https://github.com/badges/poser/pull/41#issuecomment-662385869)

@antonkomarev of course! it was already in my mind to wait for your refactoring! ;)

**Comment by antonkomarev** [comment]
> @garak @AlessandroMinoccheri what do you think about if we give up phpstan or psalm and publish this v2.0? :)

Why give up with Psalm or PHPStan?

**Comment by JellyBellyDev** [comment]
> > @garak @AlessandroMinoccheri what do you think about if we give up phpstan or psalm and publish this v2.0? :)
> 
> Why give up with Psalm or PHPStan?


unfortunately I have no time available to devote to this task and I would like to publish 2.0!  :(

**Comment by AlessandroMinoccheri** [comment]
We can open a new issue for phpstan or psalm and try to publish version 2.0 asap.

**Comment by antonkomarev** [comment]
Is there a reason why we should to hurry up to release v2? I could take a look on this issue after the #46

**Comment by JellyBellyDev** [comment]
> Is there a reason why we should to hurry up to release v2? I could take a look on this issue after the #46
> Fixes from static analysis may introduce breaking changes.

Why do I think it's going too long... that's all! :)

**Comment by antonkomarev** [comment]
I want to add all predefined colors to match shields.io website.

**Comment by JellyBellyDev** [comment]
> I want to add all predefined colors to match shields.io website.

@antonkomarev we ready to merge and tag or waiting this?

**Comment by antonkomarev** [comment]
As you wish. I could add it after the release, doesn't seem like breaking change.

**Comment by JellyBellyDev** [comment]
> As you wish. I could add it after the release, doesn't seem like breaking change.

I prefer to wait... :)

**Comment by antonkomarev** [comment]
Will look on it today evening.

**Comment by antonkomarev** [comment]
Done in #51

**Pull request merged**